packages/Discretization/test/tstInterpolation.cpp

```diff
@@ -0,0 +1,656 @@
+    std::set<PtCoord> ref_frame_set_6( ref_frame_6.begin(), ref_frame_6.end() );
+    ref_sol[pt_6] = ref_frame_set_6;
+    // Seventh point
+    PtCoord pt_7 = {{2.25, 3.75, 4.25}};
+    std::vector<PtCoord> ref_frame_7 = {{{-0.5, 0.5, -0.5}}};
+    std::set<PtCoord> ref_frame_set_7( ref_frame_7.begin(), ref_frame_7.end() );
+    ref_sol[pt_7] = ref_frame_set_7;
+    // Eigth point
+    PtCoord pt_8 = {{3.75, 3., 4.25}};
+    std::vector<PtCoord> ref_frame_8 = {{{0.5, -1., -0.5}}, {{0.5, 1., -0.5}}};
+    std::set<PtCoord> ref_frame_set_8( ref_frame_8.begin(), ref_frame_8.end() );
+    ref_sol[pt_8] = ref_frame_set_8;
+    // Nineth point
+    PtCoord pt_9 = {{3.5, 3., 4.}};
+    std::vector<PtCoord> ref_frame_9 = {
+        {{0., -1., 1.}}, {{0., 1., 1.}}, {{0., -1., -1.}}, {{0., 1., -1.}}};
+    std::set<PtCoord> ref_frame_set_9( ref_frame_9.begin(), ref_frame_9.end() );
+    ref_sol[pt_9] = ref_frame_set_9;
+    // Tenth point
+    PtCoord pt_10 = {{3., 3., 3.}};
+    std::vector<PtCoord> ref_frame_10 = {
+        {{-1., -1., -1.}}, {{-1., -1., 1.}}, {{-1., 1., 1.}},  {{1., 1., 1.}},
+        {{1., -1., -1.}},  {{1., 1., -1.}},  {{-1., 1., -1.}}, {{1., -1., 1.}}};
+    std::set<PtCoord> ref_frame_set_10( ref_frame_10.begin(),
+                                        ref_frame_10.end() );
+    ref_sol[pt_10] = ref_frame_set_10;
+
+    auto phys_points_host = Kokkos::create_mirror_view( phys_points );
+    Kokkos::deep_copy( phys_points_host, phys_points );
+    auto reference_points_host = Kokkos::create_mirror_view( reference_points );
+    Kokkos::deep_copy( reference_points_host, reference_points );
+    auto point_in_cell_host = Kokkos::create_mirror_view( point_in_cell );
+    Kokkos::deep_copy( point_in_cell_host, point_in_cell );
+    unsigned int n_points_in_cell = 0;
+    for ( unsigned int i = 0; i < point_in_cell_host.extent( 0 ); ++i )
+        if ( point_in_cell_host( i ) )
+            ++n_points_in_cell;
+    if ( comm_rank < 2 )
+    {
+        TEST_EQUALITY( n_points_in_cell, 16 );
+    }
+    else
+    {
+        TEST_EQUALITY( n_points_in_cell, 0 );
+    }
+
+    for ( unsigned int i = 0; i < phys_points.extent( 0 ); ++i )
+    {
+        if ( point_in_cell_host( i ) )
+        {
+            PtCoord pt_coord = {{phys_points_host( i )[0],
+                                 phys_points_host( i )[1],
+                                 phys_points_host( i )[2]}};
+
+            PtCoord ref_pt = {{reference_points_host( i )[0],
+                               reference_points_host( i )[1],
+                               reference_points_host( i )[2]}};
+
+            TEST_EQUALITY( ref_sol[pt_coord].count( ref_pt ), 1 );
+            ref_sol[pt_coord].erase( ref_pt );
+        }
+    }
+}
+
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( Interpolation, two_topo_two_dim, DeviceType )
+{
+    // Test a mesh of made of Quadrilateral<4> and Triangle<3>
+    // __________
+    // |___/\___|
+    // |___\/___|
```

Comment:
Come on you can do better!

Reply:
You have too high expectations. Not everybody is the Michael Angelo of ascii art like you!